fvtest/compilertriltest/SimplifierFoldAndTest.cpp

```diff
@@ -117,3 +118,46 @@ TEST_F(SimplifierFoldAndTest, FoldHappens) {
     EXPECT_EQ(0ll, entry_point(-9));
     EXPECT_EQ(0ll, entry_point(2147483647));
 }
+
+
+/**
+ * Test Fixture for SimplifierFoldAndTest that 
+ * selects only the relevant opts for the test case
+ */
+class MockOptimizationTest : public TRTest::JitOptTest
+   {
+
+   public:
+   MockOptimizationTest()
+      {
+      /*
+       * By adding a non-simplifier opt, we make sure the 
+       * simplifier doesn't run. 
+       */
+      addOptimization(OMR::trivialDeadTreeRemoval);
+      }
+
+   };
+
+/*
+ * Given the above, if we disable the simplifier
+ * the same trees should _not_ fold. This test case
+ * ensures that the mock optimizer support is working. 
+ *
+ * Coupling note: This is assuming the default optimization
+ * strategy contains the simplifier. If it did not, this test
+ * case would be invalid
+ */
+TEST_F(MockOptimizationTest, FoldDoesntHappen) {
```

Comment:
If I understand correctly, this test is verifying that the mock optimizer is being used and is working. Would it not be better than to put this test in it's own file? This would allow more extensive testing of the mock optimizer later on.

Reply:
I realized after I went home last night that calling it "mock optimizer" is the wrong phrase -- it's a custom strategy; But I will split this test out. The downside is the duplication of the test case string.